### PR TITLE
Implement generic visibility / occlusion update step

### PR DESCRIPTION
**Pull request by rdumusc**
* Currently used by DynamicTexture, TODO other content types
* ContentWindow decorations (titles, border) are added outside the windows
  and no longer alter their positions

### CODE REVIEW

File: dc/core/VisiblityHelper.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*********************************************************************/
```
Comment (tribal-tec) [inline dc/core/VisiblityHelper.cpp]: typo for the filename (missing 'i' between 'b' and 'l')

File: dc/core/VisiblityHelper.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    : _displayGroup( displayGroup )
+    , _visibleArea( visibleArea )
+{}
+
+QRectF cutOverlap( const QRectF& window, const QRectF& other )
+{
+    if( other.contains( window ))
+        return QRectF();
+
+    QRectF overlap = window.intersected( other );
```
Comment (tribal-tec) [inline dc/core/VisiblityHelper.cpp]: const

File: dc/core/VisiblityHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    }
+
+    return window;
+}
+
+QRectF VisiblityHelper::getVisibleArea( const ContentWindow& window ) const
+{
+    const QRectF& windowCoords = window.getDisplayCoordinates();
+
+    QRectF area = windowCoords.intersected( _visibleArea );
```
Comment (tribal-tec) [inline dc/core/VisiblityHelper.cpp]: const

File: tests/cpp/core/VisibilityHelperTests.cpp [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+namespace
+{
+const QSize groupSize( 800, 600 );
+const QSize size( 200, 200 );
+const QRect viewRect( 0, 0, groupSize.width() / 2, groupSize.height( ));
+const QRect centeredViewRect( 350, 250, 100, 100 );
+}
+
+struct F {
```
Comment (tribal-tec) [inline tests/cpp/core/VisibilityHelperTests.cpp]: name it Fixture?

**Comment by tribal-tec** [comment]
+1 otherwise

File: dc/core/VisiblityHelper.cpp (hunk 1 above)
Comment (rdumusc) [inline dc/core/VisiblityHelper.cpp]: argh! done!

File: dc/core/VisiblityHelper.cpp (hunk 2 above)
Comment (rdumusc) [inline dc/core/VisiblityHelper.cpp]: done

File: dc/core/VisiblityHelper.cpp (hunk 3 above)
Comment (rdumusc) [inline dc/core/VisiblityHelper.cpp]: no, modified multiple times in the for loop

File: tests/cpp/core/VisibilityHelperTests.cpp (hunk 4 above)
Comment (rdumusc) [inline tests/cpp/core/VisibilityHelperTests.cpp]: done

**Comment by rdumusc** [comment]
updated

**Pull request merged**